File: Accordinez.java
import java.awt.BorderLayout;
import java.awt.Color;
import java.awt.Dimension;
import java.awt.Font;
import java.awt.event.KeyAdapter;
import java.awt.event.KeyEvent;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;
import java.util.Set;
import javax.sound.sampled.AudioFormat;
import javax.sound.sampled.AudioSystem;
import javax.sound.sampled.LineUnavailableException;
import javax.sound.sampled.SourceDataLine;
import javax.swing.BorderFactory;
import javax.swing.JFrame;
import javax.swing.JLabel;
import javax.swing.JPanel;
import javax.swing.SwingConstants;
import javax.swing.SwingUtilities;

public final class Accordinez {
    private static final Map<Character, Bank> CHORD_SCALE_BANKS = new LinkedHashMap<>();
    private static final Map<Character, Integer> RIGHT_HAND_KEYS = new LinkedHashMap<>();
    private static final Set<Character> BANK_KEYS = new HashSet<>();

    private static final char PITCH_KEY = 't';
    private static final char OCTAVE_DOWN_KEY = 'c';
    private static final char OCTAVE_UP_KEY = 'v';

    private static final int SAMPLE_RATE = 48_000;
    private static final int BLOCK_SIZE = 256;
    private static final double MASTER_GAIN = 0.2;
    private static final int MIN_MIDI = 0;
    private static final int MAX_MIDI = 127;

    static {
        CHORD_SCALE_BANKS.put('a', new Bank("C ionian", new int[] {60, 62, 64, 65, 67, 69, 71}));
        CHORD_SCALE_BANKS.put('w', new Bank("D dorian", new int[] {62, 64, 65, 67, 69, 71, 72}));
        CHORD_SCALE_BANKS.put('s', new Bank("E phrygian", new int[] {64, 65, 67, 69, 71, 72, 74}));
        CHORD_SCALE_BANKS.put('e', new Bank("F lydian", new int[] {65, 67, 69, 71, 72, 74, 76}));
        CHORD_SCALE_BANKS.put('d', new Bank("G mixolydian", new int[] {67, 69, 71, 72, 74, 76, 77}));
        CHORD_SCALE_BANKS.put('r', new Bank("A aeolian", new int[] {69, 71, 72, 74, 76, 77, 79}));
        CHORD_SCALE_BANKS.put('f', new Bank("B locrian", new int[] {71, 72, 74, 76, 77, 79, 81}));
        CHORD_SCALE_BANKS.put('g', new Bank("E custom", new int[] {64, 65, 68, 69, 71, 72, 74}));
        BANK_KEYS.addAll(CHORD_SCALE_BANKS.keySet());

        RIGHT_HAND_KEYS.put('j', 0);
        RIGHT_HAND_KEYS.put('i', 1);
        RIGHT_HAND_KEYS.put('k', 2);
        RIGHT_HAND_KEYS.put('o', 3);
        RIGHT_HAND_KEYS.put('l', 4);
        RIGHT_HAND_KEYS.put('p', 5);
        RIGHT_HAND_KEYS.put(';', 6);
    }

    private final Object lock = new Object();
    private final Set<Character> heldRightKeys = new HashSet<>();
    private final Set<Character> heldControlKeys = new HashSet<>();
    private final Map<Character, Voice> voicesByRightKey = new HashMap<>();

    private char activeBankKey = 'a';
    private int octaveOffset = 0;
    private boolean pitchShiftActive = false;
    private volatile boolean running = true;

    private SineSynth synth;
    private JLabel statusLabel;

    public static void main(String[] args) {
        SwingUtilities.invokeLater(() -> {
            try {
                new Accordinez().start();
            } catch (LineUnavailableException ex) {
                System.err.println("Could not open audio output: " + ex.getMessage());
                System.exit(1);
            }
        });
    }

    private void start() throws LineUnavailableException {
        synth = new SineSynth();
        synth.start();

        JFrame frame = new JFrame("Accordinez");
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        frame.setMinimumSize(new Dimension(440, 220));
        frame.setContentPane(buildPanel());
        frame.addKeyListener(new KeyHandler());
        frame.addWindowListener(new java.awt.event.WindowAdapter() {
            @Override
            public void windowClosing(java.awt.event.WindowEvent event) {
                shutdown();
            }
        });
        frame.pack();
        frame.setLocationRelativeTo(null);
        frame.setVisible(true);
        frame.requestFocusInWindow();

        Runtime.getRuntime().addShutdownHook(new Thread(this::shutdown));
        debug("ready");
    }

    private JPanel buildPanel() {
        JPanel panel = new JPanel(new BorderLayout());
        panel.setBackground(new Color(24, 25, 27));
        panel.setBorder(BorderFactory.createEmptyBorder(36, 36, 36, 36));
        panel.setFocusable(true);

        JLabel title = new JLabel("Accordinez", SwingConstants.CENTER);
        title.setForeground(new Color(238, 238, 232));
        title.setFont(new Font(Font.SANS_SERIF, Font.BOLD, 32));

        statusLabel = new JLabel("", SwingConstants.CENTER);
        statusLabel.setForeground(new Color(174, 183, 191));
        statusLabel.setFont(new Font(Font.MONOSPACED, Font.PLAIN, 15));

        panel.add(title, BorderLayout.CENTER);
        panel.add(statusLabel, BorderLayout.SOUTH);
        return panel;
    }

    private static double midiToFreq(int midiNote) {
        return 440.0 * Math.pow(2.0, (midiNote - 69) / 12.0);
    }

    private int clampMidi(int midiNote) {
        return Math.max(MIN_MIDI, Math.min(MAX_MIDI, midiNote));
    }

    private int effectiveNote(int baseMidi) {
        return clampMidi(baseMidi + (pitchShiftActive ? 1 : 0));
    }

    private void updateStatus() {
        if (statusLabel == null) {
            return;
        }
        Bank bank = CHORD_SCALE_BANKS.get(activeBankKey);
        String text = String.format("bank %c:%s   oct %+d   shift %s   voices %d",
                activeBankKey, bank.name, octaveOffset, pitchShiftActive ? "on" : "off", voicesByRightKey.size());
        statusLabel.setText(text);
    }

    private void debug(String label) {
        StringBuilder active = new StringBuilder();
        List<Character> keys = new ArrayList<>(voicesByRightKey.keySet());
        keys.sort(Character::compareTo);
        for (char key : keys) {
            Voice voice = voicesByRightKey.get(key);
            if (active.length() > 0) {
                active.append(", ");
            }
            active.append(key)
                    .append(":")
                    .append(voice.currentMidi)
                    .append("(")
                    .append(voice.bankKeyAtNoteOn)
                    .append(")");
        }
        if (active.length() == 0) {
            active.append("-");
        }

        Bank bank = CHORD_SCALE_BANKS.get(activeBankKey);
        System.out.printf("[%s] bank=%c:%s oct=%+d shift=%s voices=%s%n",
                label, activeBankKey, bank.name, octaveOffset, pitchShiftActive, active);
        updateStatus();
    }

    private void startVoice(char rightKey) {
        if (voicesByRightKey.containsKey(rightKey)) {
            return;
        }

        int degreeIdx = RIGHT_HAND_KEYS.get(rightKey);
        int baseMidi = CHORD_SCALE_BANKS.get(activeBankKey).degrees[degreeIdx] + octaveOffset;
        baseMidi = clampMidi(baseMidi);
        int currentMidi = effectiveNote(baseMidi);

        Voice voice = new Voice(rightKey, activeBankKey, degreeIdx, baseMidi, currentMidi,
                midiToFreq(baseMidi), midiToFreq(currentMidi));
        voicesByRightKey.put(rightKey, voice);
        debug("note_on " + rightKey + "->" + currentMidi);
    }

    private void stopVoice(char rightKey) {
        Voice voice = voicesByRightKey.remove(rightKey);
        if (voice == null) {
            return;
        }
        debug("note_off " + rightKey + "->" + voice.currentMidi);
    }

    private void retuneAllActiveVoices() {
        for (Voice voice : voicesByRightKey.values()) {
            int targetMidi = effectiveNote(voice.baseMidi);
            voice.currentMidi = targetMidi;
            voice.currentFreq = midiToFreq(targetMidi);
        }
        debug("retune_all");
    }

    private void handleBankSelect(char ch) {
        if (ch == activeBankKey) {
            return;
        }
        activeBankKey = ch;
        debug("bank_switch " + ch);
    }

    private void handlePitchKeyDown() {
        if (pitchShiftActive) {
            return;
        }
        pitchShiftActive = true;
        retuneAllActiveVoices();
    }

    private void handlePitchKeyUp() {
        if (!pitchShiftActive) {
            return;
        }
        pitchShiftActive = false;
        retuneAllActiveVoices();
    }

    private void allNotesOff() {
        voicesByRightKey.clear();
        heldRightKeys.clear();
        debug("all_notes_off");
    }

    private void onPress(char rawChar) {
        char ch = Character.toLowerCase(rawChar);
        synchronized (lock) {
            if (BANK_KEYS.contains(ch) || ch == PITCH_KEY || ch == OCTAVE_DOWN_KEY || ch == OCTAVE_UP_KEY) {
                if (heldControlKeys.contains(ch)) {
                    return;
                }
                heldControlKeys.add(ch);
            }

            if (BANK_KEYS.contains(ch)) {
                handleBankSelect(ch);
                return;
            }

            if (ch == PITCH_KEY) {
                handlePitchKeyDown();
                return;
            }

            if (ch == OCTAVE_DOWN_KEY) {
                octaveOffset -= 12;
                debug("octave_down");
                return;
            }

            if (ch == OCTAVE_UP_KEY) {
                octaveOffset += 12;
                debug("octave_up");
                return;
            }

            if (RIGHT_HAND_KEYS.containsKey(ch)) {
                if (heldRightKeys.contains(ch)) {
                    return;
                }
                heldRightKeys.add(ch);
                startVoice(ch);
            }
        }
    }

    private void onRelease(char rawChar) {
        char ch = Character.toLowerCase(rawChar);
        synchronized (lock) {
            if (ch == PITCH_KEY) {
                heldControlKeys.remove(ch);
                handlePitchKeyUp();
                return;
            }

            if (ch == OCTAVE_DOWN_KEY || ch == OCTAVE_UP_KEY || BANK_KEYS.contains(ch)) {
                heldControlKeys.remove(ch);
                return;
            }

            if (RIGHT_HAND_KEYS.containsKey(ch)) {
                heldRightKeys.remove(ch);
                stopVoice(ch);
            }
        }
    }

    private void shutdown() {
        if (!running) {
            return;
        }
        running = false;
        synchronized (lock) {
            allNotesOff();
        }
        if (synth != null) {
            synth.stop();
        }
    }

    private final class KeyHandler extends KeyAdapter {
        @Override
        public void keyPressed(KeyEvent event) {
            if (event.getKeyCode() == KeyEvent.VK_ESCAPE) {
                shutdown();
                System.exit(0);
                return;
            }

            char ch = event.getKeyChar();
            if (ch != KeyEvent.CHAR_UNDEFINED) {
                onPress(ch);
            }
        }

        @Override
        public void keyReleased(KeyEvent event) {
            char ch = event.getKeyChar();
            if (ch != KeyEvent.CHAR_UNDEFINED) {
                onRelease(ch);
            }
        }
    }

    private final class SineSynth {
        private final SourceDataLine line;
        private final Thread thread;

        SineSynth() throws LineUnavailableException {
            AudioFormat format = new AudioFormat(SAMPLE_RATE, 16, 1, true, false);
            line = AudioSystem.getSourceDataLine(format);
            line.open(format, BLOCK_SIZE * 4);
            thread = new Thread(this::runAudio, "accordinez-audio");
            thread.setDaemon(true);
        }

        void start() {
            line.start();
            thread.start();
        }

        void stop() {
            line.stop();
            line.flush();
            line.close();
        }

        private void runAudio() {
            byte[] buffer = new byte[BLOCK_SIZE * 2];
            while (running) {
                render(buffer);
                line.write(buffer, 0, buffer.length);
            }
        }

        private void render(byte[] buffer) {
            synchronized (lock) {
                List<Voice> voices = new ArrayList<>(voicesByRightKey.values());
                if (voices.isEmpty()) {
                    for (int i = 0; i < buffer.length; i++) {
                        buffer[i] = 0;
                    }
                    return;
                }

                double perVoiceGain = MASTER_GAIN / Math.sqrt(voices.size());
                int byteIndex = 0;
                for (int frame = 0; frame < BLOCK_SIZE; frame++) {
                    double mix = 0.0;
                    for (Voice voice : voices) {
                        mix += perVoiceGain * Math.sin(voice.phase);
                        double inc = (2.0 * Math.PI * voice.currentFreq) / SAMPLE_RATE;
                        voice.phase = (voice.phase + inc) % (2.0 * Math.PI);
                    }
                    mix = Math.max(-1.0, Math.min(1.0, mix));
                    short sample = (short) Math.round(mix * Short.MAX_VALUE);
                    buffer[byteIndex++] = (byte) (sample & 0xff);
                    buffer[byteIndex++] = (byte) ((sample >>> 8) & 0xff);
                }
            }
        }
    }

    private static final class Bank {
        final String name;
        final int[] degrees;

        Bank(String name, int[] degrees) {
            this.name = name;
            this.degrees = degrees;
        }
    }

    private static final class Voice {
        final char rightKey;
        final char bankKeyAtNoteOn;
        final int degreeIdx;
        final int baseMidi;
        final double baseFreq;
        int currentMidi;
        double currentFreq;
        double phase;

        Voice(char rightKey, char bankKeyAtNoteOn, int degreeIdx, int baseMidi,
                int currentMidi, double baseFreq, double currentFreq) {
            this.rightKey = rightKey;
            this.bankKeyAtNoteOn = bankKeyAtNoteOn;
            this.degreeIdx = degreeIdx;
            this.baseMidi = baseMidi;
            this.currentMidi = currentMidi;
            this.baseFreq = baseFreq;
            this.currentFreq = currentFreq;
            this.phase = 0.0;
        }
    }
}

File: accordinez.py
#!/usr/bin/env python3
"""
Accordinez prototype: polyphonic sine-wave synth controlled by computer keyboard.

Dependencies:
    pip install pynput sounddevice numpy

Usage:
    python accordinez.py
"""

from __future__ import annotations

import signal
import os
import sys
import threading
from dataclasses import dataclass
from typing import Dict, Optional, Set

import numpy as np
import sounddevice as sd

try:
    from pynput import keyboard
except ImportError as exc:
    keyboard = None
    KEYBOARD_IMPORT_ERROR: Optional[ImportError] = exc
else:
    KEYBOARD_IMPORT_ERROR = None


# Global chord-scale bank definition (requested for easy top-level access).
CHORD_SCALE_BANKS = {
    "a": {"name": "C ionian", "degrees": [60, 62, 64, 65, 67, 69, 71]},
    "w": {"name": "D dorian", "degrees": [62, 64, 65, 67, 69, 71, 72]},
    "s": {"name": "E phrygian", "degrees": [64, 65, 67, 69, 71, 72, 74]},
    "e": {"name": "F lydian", "degrees": [65, 67, 69, 71, 72, 74, 76]},
    "d": {"name": "G mixolydian", "degrees": [67, 69, 71, 72, 74, 76, 77]},
    "r": {"name": "A aeolian", "degrees": [69, 71, 72, 74, 76, 77, 79]},
    "f": {"name": "B locrian", "degrees": [71, 72, 74, 76, 77, 79, 81]},
    "g": {"name": "E custom", "degrees": [64, 65, 68, 69, 71, 72, 74]},
}

RIGHT_HAND_KEYS = {
    "j": 0,
    "i": 1,
    "k": 2,
    "o": 3,
    "l": 4,
    "p": 5,
    ";": 6,
}

BANK_KEYS = set(CHORD_SCALE_BANKS.keys())
PITCH_KEY = "t"
OCTAVE_DOWN_KEY = "c"
OCTAVE_UP_KEY = "v"

SAMPLE_RATE = 48000
BLOCK_SIZE = 256
MASTER_GAIN = 0.2
MIN_MIDI = 0
MAX_MIDI = 127


@dataclass
class Voice:
    right_key: str
    bank_key_at_note_on: str
    degree_idx: int
    base_midi: int
    current_midi: int
    base_freq: float
    current_freq: float
    phase: float = 0.0


def midi_to_freq(midi_note: int) -> float:
    return 440.0 * (2.0 ** ((midi_note - 69) / 12.0))


class SineSynth:
    def __init__(self, get_voices_cb, lock: threading.Lock) -> None:
        self.get_voices_cb = get_voices_cb
        self.lock = lock
        self.stream = sd.OutputStream(
            samplerate=SAMPLE_RATE,
            channels=1,
            dtype="float32",
            blocksize=BLOCK_SIZE,
            callback=self.audio_callback,
        )

    def start(self) -> None:
        self.stream.start()

    def stop(self) -> None:
        self.stream.stop()
        self.stream.close()

    def audio_callback(self, outdata, frames, _time, _status) -> None:
        with self.lock:
            voices = list(self.get_voices_cb().values())

            if not voices:
                outdata.fill(0)
                return

            mix = np.zeros(frames, dtype=np.float32)
            per_voice_gain = MASTER_GAIN / max(1.0, np.sqrt(len(voices)))

            for voice in voices:
                inc = (2.0 * np.pi * voice.current_freq) / SAMPLE_RATE
                phase_series = voice.phase + inc * np.arange(frames, dtype=np.float32)
                wave = np.sin(phase_series, dtype=np.float32)
                mix += per_voice_gain * wave
                voice.phase = float((voice.phase + inc * frames) % (2.0 * np.pi))

            np.clip(mix, -1.0, 1.0, out=mix)
            outdata[:, 0] = mix


class Accordinez:
    def __init__(self) -> None:
        self.lock = threading.Lock()

        self.active_bank_key: str = "a"
        self.octave_offset: int = 0
        self.pitch_shift_active: bool = False

        self.held_right_keys: Set[str] = set()
        self.held_control_keys: Set[str] = set()
        self.voices_by_right_key: Dict[str, Voice] = {}

        self.listener: Optional[keyboard.Listener] = None
        self.running = True

        self.synth = SineSynth(self._voices_ref, self.lock)
        self.synth_started = False

    def _voices_ref(self) -> Dict[str, Voice]:
        return self.voices_by_right_key

    def key_to_char(self, key: keyboard.Key | keyboard.KeyCode) -> Optional[str]:
        if isinstance(key, keyboard.KeyCode):
            if key.char is None:
                return None
            return key.char.lower()
        return None

    def clamp_midi(self, midi_note: int) -> int:
        return max(MIN_MIDI, min(MAX_MIDI, midi_note))

    def effective_note(self, base_midi: int) -> int:
        return self.clamp_midi(base_midi + (1 if self.pitch_shift_active else 0))

    def debug(self, label: str) -> None:
        active = ", ".join(
            f"{k}:{v.current_midi}({v.bank_key_at_note_on})"
            for k, v in sorted(self.voices_by_right_key.items())
        )
        active = active or "-"
        bank_name = CHORD_SCALE_BANKS[self.active_bank_key]["name"]
        print(
            f"[{label}] bank={self.active_bank_key}:{bank_name} "
            f"oct={self.octave_offset:+} shift={self.pitch_shift_active} voices={active}"
        )

    def start_voice(self, right_key: str) -> None:
        if right_key in self.voices_by_right_key:
            return

        degree_idx = RIGHT_HAND_KEYS[right_key]
        base_midi = CHORD_SCALE_BANKS[self.active_bank_key]["degrees"][degree_idx] + self.octave_offset
        base_midi = self.clamp_midi(base_midi)

        current_midi = self.effective_note(base_midi)
        voice = Voice(
            right_key=right_key,
            bank_key_at_note_on=self.active_bank_key,
            degree_idx=degree_idx,
            base_midi=base_midi,
            current_midi=current_midi,
            base_freq=midi_to_freq(base_midi),
            current_freq=midi_to_freq(current_midi),
        )
        self.voices_by_right_key[right_key] = voice
        self.debug(f"note_on {right_key}->{current_midi}")

    def stop_voice(self, right_key: str) -> None:
        voice = self.voices_by_right_key.pop(right_key, None)
        if voice is None:
            return
        self.debug(f"note_off {right_key}->{voice.current_midi}")

    def retune_all_active_voices(self) -> None:
        for voice in self.voices_by_right_key.values():
            target_midi = self.effective_note(voice.base_midi)
            voice.current_midi = target_midi
            voice.current_freq = midi_to_freq(target_midi)
        self.debug("retune_all")

    def handle_bank_select(self, ch: str) -> None:
        if ch == self.active_bank_key:
            return
        self.active_bank_key = ch
        # Lazy update: do not retune sustained voices.
        self.debug(f"bank_switch {ch}")

    def handle_pitch_key_down(self) -> None:
        if self.pitch_shift_active:
            return
        self.pitch_shift_active = True
        self.retune_all_active_voices()

    def handle_pitch_key_up(self) -> None:
        if not self.pitch_shift_active:
            return
        self.pitch_shift_active = False
        self.retune_all_active_voices()

    def all_notes_off(self) -> None:
        self.voices_by_right_key.clear()
        self.held_right_keys.clear()
        self.debug("all_notes_off")

    def on_press(self, key: keyboard.Key | keyboard.KeyCode) -> None:
        ch = self.key_to_char(key)
        if ch is None:
            return

        with self.lock:
            if ch in BANK_KEYS | {PITCH_KEY, OCTAVE_DOWN_KEY, OCTAVE_UP_KEY}:
                if ch in self.held_control_keys:
                    return
                self.held_control_keys.add(ch)

            if ch in BANK_KEYS:
                self.handle_bank_select(ch)
                return

            if ch == PITCH_KEY:
                self.handle_pitch_key_down()
                return

            if ch == OCTAVE_DOWN_KEY:
                self.octave_offset -= 12
                self.debug("octave_down")
                return

            if ch == OCTAVE_UP_KEY:
                self.octave_offset += 12
                self.debug("octave_up")
                return

            if ch in RIGHT_HAND_KEYS:
                if ch in self.held_right_keys:
                    return
                self.held_right_keys.add(ch)
                self.start_voice(ch)

    def on_release(self, key: keyboard.Key | keyboard.KeyCode) -> None:
        ch = self.key_to_char(key)
        if ch is None:
            return

        with self.lock:
            if ch in self.held_control_keys:
                self.held_control_keys.remove(ch)

            if ch == PITCH_KEY:
                self.handle_pitch_key_up()
                return

            if ch in RIGHT_HAND_KEYS:
                if ch in self.held_right_keys:
                    self.held_right_keys.remove(ch)
                self.stop_voice(ch)

    def shutdown(self) -> None:
        with self.lock:
            self.running = False
            self.all_notes_off()

        if self.listener is not None:
            self.listener.stop()

        if self.synth_started:
            self.synth.stop()
            self.synth_started = False

    def run(self) -> None:
        if keyboard is None:
            print("Accordinez cannot start keyboard input.")
            print("The pynput keyboard backend failed to initialize:")
            print(f"  {KEYBOARD_IMPORT_ERROR}")
            if sys.platform.startswith("linux"):
                print()
                print("Linux note: Accordinez currently uses pynput's global keyboard listener.")
                print("That backend requires an X11 session. It usually cannot read keys under Wayland.")
                print("Try logging into an Xorg/X11 session, or run from a terminal with DISPLAY set.")
            return

        if sys.platform.startswith("linux"):
            session_type = os.environ.get("XDG_SESSION_TYPE", "").lower()
            if session_type == "wayland" or (os.environ.get("WAYLAND_DISPLAY") and not os.environ.get("DISPLAY")):
                print("Warning: this looks like a Wayland Linux session.")
                print("pynput keyboard listeners often cannot receive key events under Wayland.")
                print("If keys do nothing, log into an Xorg/X11 session and run Accordinez there.")
                print()

        print("Accordinez running (polyphonic sine-wave synth).")
        print("Left hand: a/w/s/e/d/r/f/g (bank), t (+1 semitone hold), c/v (oct down/up).")
        print("Right hand: j i k o l p ; mapped to degrees 1..7.")
        print("Press Ctrl+C to exit.")

        self.synth.start()
        self.synth_started = True
        self.listener = keyboard.Listener(on_press=self.on_press, on_release=self.on_release)
        self.listener.start()
        self.listener.join()


def main() -> int:
    app = Accordinez()

    def _signal_handler(_sig: int, _frame: object) -> None:
        app.shutdown()
        raise KeyboardInterrupt

    signal.signal(signal.SIGINT, _signal_handler)
    signal.signal(signal.SIGTERM, _signal_handler)

    try:
        app.run()
    except KeyboardInterrupt:
        pass
    finally:
        app.shutdown()
    return 0


if __name__ == "__main__":
    sys.exit(main())
